### ISSUE TITLE
 The logger issue in meshkit@0.5.2

### ISSUE THREAD

**Issue by Aisuko** [meshery/meshkit#148]
**Description**

Hi, team. Please help to confirm that meshkit@0.5.2 issue with the logger

https://github.com/meshery/meshery-linkerd/runs/4887907143?check_suite_focus=true

https://github.com/meshery/meshery-linkerd/runs/4887907143?check_suite_focus=true

```
# github.com/layer5io/meshkit/logger
167
Error: ../../../go/pkg/mod/github.com/layer5io/meshkit@v0.5.2/logger/controller.go:23:9: cannot use &Controller{...} (type *Controller) as type logr.Logger in return argument
168
Error: ../../../go/pkg/mod/github.com/layer5io/meshkit@v0.5.2/logger/controller.go:42:2: cannot use c (type *Controller) as type logr.Logger in return argument
169
Error: ../../../go/pkg/mod/github.com/layer5io/meshkit@v0.5.2/logger/controller.go:47:2: cannot use c (type *Controller) as type logr.Logger in return argument
170
Error: ../../../go/pkg/mod/github.com/layer5io/meshkit@v0.5.2/logger/controller.go:52:2: cannot use c (type *Controller) as type logr.Logger in return argument
```

**Expected Behavior**

**Screenshots**

**Environment:**
 - OS: [e.g. Ubuntu]
 - Browser: [e.g. Chrome, Safari]
 - Version: [e.g. 22]
 - Device: [e.g. laptop, iPhone 8]

---
[Optional] **To Reproduce**

[Optional] **Additional Context**

**Comment by Aisuko** [comment]
https://github.com/meshery/meshery-operator/pull/257

**Comment by leecalcote** [comment]
Does @piyushsingariya's https://github.com/meshery/meshkit/pull/146 fix this?

**Comment by piyushsingariya** [comment]
I know this issue, I lowered one dependency for fixing this in one of my PR, but #146 isn't that PR.

**Comment by piyushsingariya** [comment]
The latest changes in the master branch don't have this issue @Aisuko you check.

**Comment by Aisuko** [comment]
> The latest changes in the master branch don't have this issue @Aisuko you check.

Noticed and thanks.

**Comment by Aisuko** [comment]
Will close after checked it.

**Comment by jbyers19** [comment]
@Aisuko, just saw this now. This should be fixed with https://github.com/meshery/meshkit/pull/155, which was merged a couple days ago.

**Comment by jbyers19** [comment]
@Aisuko, ended up having to revert #155 (revert PR: #158) due to conflicts with other repos. If this is a blocker for you, you can pull the just the logger changes from #155.

If you don't need it now, I'm planning to get all the Meshery components updated to go v1.17 after Meshery v0.6.0 is released.

**Comment by jbyers19** [comment]
@Aisuko, the logger changes have been merged and released as part of `v0.5.14`. That should fix this issue.

**Comment by Aisuko** [comment]
> @Aisuko, the logger changes have been merged and released as part of `v0.5.14`. That should fix this issue.

Gotcha. Thank you.

**Comment by Aisuko** [comment]
Hi @jbyers19 , feel free help close this issue.
